pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/InMemorySendingMailbox.java

```diff
@@ -68,7 +68,11 @@ public void send(MseBlock.Data data)
   @Override
   public void send(MseBlock.Eos block, List<DataBuffer> serializedStats)
       throws IOException, TimeoutException {
-    sendPrivate(block, serializedStats);
+    try {
+      sendPrivate(block, serializedStats);
+    } finally {
+      _isTerminated = true;
```

Comment:
Should we put this in the `finally` block? This could prevent error block being set in `cancel()` when `sendPrivate()` throws exception

Reply:
Good catch. I'll change it